From f5ae6f0113bcf95f61340c960810bd0c2dee2655 Mon Sep 17 00:00:00 2001
From: Stephan Gerhold <stephan@gerhold.net>
Date: Wed, 30 Jul 2025 16:17:46 +0200
Subject: [PATCH 15/16] arm64: dts: qcom: x1-el2: Add qcom,broken-reset for
 remoteprocs

The remoteproc PAS firmware interface does not work properly when running
bare-metal without hypervisor in EL2. It's possible to authenticate and
start new firmware, but the remoteproc is never actually brought out of
reset. On X1E this affects all remoteprocs: ADSP and CDSP.

Mark both with "qcom,broken-reset" to make the driver skip trying to reset
the remoteproc. We can still attach to them if they are already running
during boot. On X1E, the "lite" ADSP firmware should be always running
already during boot, and it provides at least battery status/charging as
well as USB-C detection for external displays.

Signed-off-by: Stephan Gerhold <stephan@gerhold.net>
Signed-off-by: Abel Vesa <abel.vesa@oss.qualcomm.com>
---
 arch/arm64/boot/dts/qcom/x1-el2.dtso | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/x1-el2.dtso b/arch/arm64/boot/dts/qcom/x1-el2.dtso
index ee006742d..4b3dceae3 100644
--- a/arch/arm64/boot/dts/qcom/x1-el2.dtso
+++ b/arch/arm64/boot/dts/qcom/x1-el2.dtso
@@ -54,10 +54,12 @@ &pcie_smmu {
 
 &remoteproc_adsp {
 	iommus = <&apps_smmu 0x1000 0x80>;
+	qcom,broken-reset;
 };
 
 &remoteproc_cdsp {
 	iommus = <&apps_smmu 0x0c00 0x0>;
+	qcom,broken-reset;
 };
 
 /*
